dev-docs/RFCs/v5.0/viewport-animation-rfc.md

```diff
@@ -60,11 +60,23 @@ viewportAnimationDuration: (Number) Animation duration in milliseconds, default
 
 1. **animationDuration** {Number, default: 0} : Animation duration in milliseconds, default value disables animation.
 
-2. **easingFunction** {Function, default: t => t} : Easing function that can be used to achieve effects like "Ease-In-Cubic", "Ease-Out-Cubic", etc. Default value performs Linear easing. (list of sample easing functions: http://easings.net/)
+2. **animaitonEasingFunction** {Function, default: t => t} : Easing function that can be used to achieve effects like "Ease-In-Cubic", "Ease-Out-Cubic", etc. Default value performs Linear easing. (list of sample easing functions: http://easings.net/)
 
-3. **animationFunction** {Function, default: Linear} : Function that gets called for each animation step to calculated animated viewport. This function takes start, end viewports and animation step, and returns animated viewport. We provide couple of utility functions, `viewportLinearAnimation` and `viewportFlyToAnimation`. By default `viewportLinearAnimation` is used where all viewport props are linearly animated. `viewportFlyToAnimation` animates viewports similar to MapBox `flyTo` API, this is pretty useful when camera center changes. But a user can provide any function for this prop to perform custom viewport animations.
+3. **animationFunction** {Function, default: `viewportLinearAnimation`} : Function that gets called for each animation step to calculated animated viewport. It takes start, end viewports and animation step, returns animated viewport. We provide couple of utility functions, `viewportLinearAnimation` and `viewportFlyToAnimation`. By default `viewportLinearAnimation` is used where all viewport props are linearly animated. `viewportFlyToAnimation` animates viewports similar to MapBox `flyTo` API, this is pretty useful when camera center changes. But a user can provide any function for this prop to perform custom viewport animations.
+
+4. **animationFreeze** {bool, optional} : When set to `true` and if `animationDuration` not equal to 0, any following viewport updates are ignored until animation resulting this update is finished. This prop has no effect if `animationDuration` is 0. When it is set to `false` or not set, any following viewport updates will stop the current animation and update viewport according to other props.
+
+5. **animationCompleteOnUpdate** {bool, false} : This prop can be used to control how to end current animation, when new animation requested before it is completed, following table describes in more details.
```

Comment:
animationCompleteOnUpdate is kind of confusing, understand that it is for user to define how we connect two animations or end animation, but hard to remember all the settings, recommend we go the following approach.
1. In the first case ideally two animations should be always smoothly blended into a single animation, adding a new API to introduce frame dropping is not good. Animation blending is not easy, don't need implement for now.
2. In the second case we can define an API animationCompleteOnCancel(bool), which define how we end the animation when animation cancelling happens, jump to destination when bool is true.
3. If user wants to introduce "frame dropping" to connect two animations, they can always call animationCompleteOnCancel(true) and start a new animation.

Reply:
- AFAIK `Animation blending` is lot of effort with little gain, if there are any use cases we can always add it later.
- Yeah, I agree, these prop names are confusing, I remove `4` and `5` and just define 3 constants as Ib suggested below.